docker/freebsd.sh

```diff
@@ -38,9 +82,9 @@ latest_freebsd() {
     local releases=
     local max_release=
 
-    response=$(curl --silent --list-only --location "${mirror}/${BSD_ARCH}/" | grep RELEASE)
+    response=$(curl --retry 3 -sSflL "${mirror}/${FREEBSD_ARCH}/" | grep RELEASE)
```

Comment:
Added the retry options to avoid the failures if the mirror is found but the subsequent attempt to query the server fails, like in our [workflows](https://github.com/cross-rs/cross/actions/runs/3570698725/jobs/6001977624#step:10:270).